### PR TITLE
Generate new TA dev kit for ARM32 platforms

### PR DESCRIPTION
**Pull request by vchong**
I didn't add ```CROSS_COMPILE_ta_arm32``` flag to ```.travis.yml``` since it should default correctly.
Update a couple of related docs as well.
Thanks!

### CODE REVIEW

File: documentation/build_system.md [hunk 1]
```diff
@@ -67,12 +67,13 @@ The output directory has basically the same structure as the source tree.
 For instance, assuming **ARCH=arm PLATFORM=stm**,
 `core/kernel/panic.c` will compile into `out/arm-plat-stm/core/kernel/panic.o`.
 
-However, some libraries are compiled twice: once for user mode, and once for
-kernel mode. This is because they may be used by the TEE Core as well as by
-the Trusted Applications. As a result, the `lib` source directory gives two
-build directories: `user_ta-lib` and `core-lib`.
+However, some libraries are compiled twice or thrice: once or twice for user
```
Comment (jenswi-linaro) [inline documentation/build_system.md]: s/twice or thrice/several times/

**Comment by jenswi-linaro** [comment]
with or without my comment addressed
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by vchong** [comment]
Comment addressed, rebased and tags applied.
Please merge.
Thanks!

**Pull request merged**